components/FlexibleHeader/src/private/MDCFlexibleHeaderMinMaxHeight.m

```diff
@@ -0,0 +1,139 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCFlexibleHeaderMinMaxHeight.h"
+
+#import "MDCFlexibleHeaderTopSafeArea.h"
+
+// The default maximum height for the header. Does not include the status bar height.
+static const CGFloat kFlexibleHeaderDefaultHeight = 56;
+
+@interface MDCFlexibleHeaderMinMaxHeight ()
+
+@property(nonatomic, strong) MDCFlexibleHeaderTopSafeArea *topSafeArea;
+
+@property(nonatomic) BOOL hasExplicitlySetMinHeight;
+@property(nonatomic) BOOL hasExplicitlySetMaxHeight;
+
+@end
+
+@implementation MDCFlexibleHeaderMinMaxHeight
+
+- (instancetype)initWithTopSafeArea:(MDCFlexibleHeaderTopSafeArea *)topSafeArea {
+  self = [super init];
+  if (self) {
+    self.topSafeArea = topSafeArea;
+
+    const CGFloat topSafeAreaInset = [self.topSafeArea topSafeAreaInset];
+
+    _minMaxHeightIncludesSafeArea = YES;
+    _minimumHeight = kFlexibleHeaderDefaultHeight + topSafeAreaInset;
+    _maximumHeight = _minimumHeight;
+  }
+  return self;
+}
+
+#pragma mark - Public
+
+- (void)setMinimumHeight:(CGFloat)minimumHeight {
+  _hasExplicitlySetMinHeight = YES;
+  if (_minimumHeight == minimumHeight) {
+    return;
+  }
+
+  _minimumHeight = minimumHeight;
+
+  if (_minimumHeight > self.maximumHeight) {
+    [self setMaximumHeight:_minimumHeight];
+  } else {
+    [self.delegate flexibleHeaderMinMaxHeightDidChange:self];
+  }
+}
+
+- (void)setMaximumHeight:(CGFloat)maximumHeight {
+  _hasExplicitlySetMaxHeight = YES;
+  if (_maximumHeight == maximumHeight) {
+    return;
+  }
+
+  _maximumHeight = maximumHeight;
+
+  [self.delegate flexibleHeaderMaximumHeightDidChange:self];
+
+  if (self.maximumHeight < _minimumHeight) {
+    [self setMinimumHeight:self.maximumHeight];
+  } else {
+    [self.delegate flexibleHeaderMinMaxHeightDidChange:self];
+  }
+}
+
+- (void)setMinMaxHeightIncludesSafeArea:(BOOL)minMaxHeightIncludesSafeArea {
```

Comment:
_Question:_ If a client follows:

1. `minimumHeight = 20`
2. `maximumHeight = 100`
3. `minMaxHeightIncludesSafeArea = YES`

Would they then loose those values of _20_ and _100_ and now just have their header be 56?